### PR TITLE
refactor(runner): clean runner handle

### PR DESCRIPTION
**Pull request by bojiang**
## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/_internal/runner/runner_handle/__init__.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    ) -> None:
+        pass
+
+    def run_method(
+        self,
+        method_name: str,
+        *args: t.Any,
+        **kwargs: t.Any,
+    ) -> t.Any:
+        raise StateException("Runner is not initialized")
```
Comment (parano) [inline bentoml/_internal/runner/runner_handle/__init__.py:54]: is the purpose of DummyRunnerHandle to get rid of the `run_method` on `runner` instance, and show a nice error message when using uninitialized runner?

File: bentoml/_internal/runner/runner_handle/local.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+if TYPE_CHECKING:
+    from ..runner import Runner
+    from ..runnable import Runnable
+
+
+class LocalRunnerRef(RunnerHandle):
+    _runnable: Runnable
+
+    def __init__(self, runner: Runner) -> None:  # pylint: disable=super-init-not-called
+        self._runnable = runner.runnable_class()
```
Comment (parano) [inline bentoml/_internal/runner/runner_handle/local.py:17]:
```suggestion
        self._runnable = runner.runnable_class(**runnable_init_params)
```

File: bentoml/_internal/runner/runner_handle/local.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+from typing import TYPE_CHECKING
+
+from . import RunnerHandle
+
+if TYPE_CHECKING:
+    from ..runner import Runner
+    from ..runnable import Runnable
+
+
+class LocalRunnerRef(RunnerHandle):
```
Comment (parano) [inline bentoml/_internal/runner/runner_handle/local.py:13]: This is just `LocalRunner` right? why call it a `Ref`?

File: bentoml/exceptions.py [hunk 4]
```diff
@@ -14,12 +16,20 @@ def __init__(self, message: str):
         super().__init__(message)
 
 
+class StateException(Exception):
```
Comment (parano) [inline bentoml/exceptions.py:19]: rename to `InvalidRunnerState`?

File: bentoml/_internal/runner/runner_handle/local.py (hunk 3 above)
Comment (bojiang) [reply bentoml/_internal/runner/runner_handle/local.py:13]: LocalRunnerRef
RemoteRunnerClient

File: bentoml/_internal/runner/runner_handle/__init__.py (hunk 1 above)
Comment (bojiang) [reply bentoml/_internal/runner/runner_handle/__init__.py:54]: The main purpose is to get rid of unnecessary nested function call, or repeatedly check the runner_handle

**Pull request merged**